agent/command/macsign.go

```diff
@@ -0,0 +1,209 @@
+	EntitlementsFilePath string `mapstructure:"entitlements_file" plugin:"expand"`
+
+	// Verify determines if the signature(or notarization) should be verified.
+	// Verification is only supported on MacOS. It is optional, default value if false.
+	Verify bool `mapstructure:"verify"`
+
+	// Notarize determines if the file should also be notarized after signing.
+	Notarize bool `mapstructure:"notarize"`
+
+	// BundleId is the bundle id of the artifact used during notarization.
+	// This is mandatory if notarization is requested.
+	BundleId string `mapstructure:"bundle_id"  plugin:"expand"`
+
+	// BuildVariants stores a list of MCI build variants to run the command for.
+	// If the list is empty, it runs for all build variants.
+	BuildVariants []string `mapstructure:"build_variants"`
+
+	// workDir sets the working directory relative to which s3put should look for files to upload.
+	// workDir will be empty if an absolute path is provided to the file.
+	workDir string
+
+	base
+}
+
+func macSignFactory() Command         { return &macSign{} }
+func (macSign *macSign) Name() string { return "mac.sign" }
+
+// macSign-specific implementation of ParseParams.
+func (macSign *macSign) ParseParams(params map[string]interface{}) error {
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Result: macSign,
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err := decoder.Decode(params); err != nil {
+		return errors.Wrapf(err, "error decoding %s params", macSign.Name())
+	}
+
+	return macSign.validate()
+}
+
+func (macSign *macSign) validate() error {
+	catcher := grip.NewSimpleCatcher()
+
+	// make sure the command params are valid
+	if macSign.KeyId == "" {
+		catcher.Add(errors.New("key_id cannot be blank"))
+	}
+	if macSign.Secret == "" {
+		catcher.Add(errors.New("secret cannot be blank"))
+	}
+	if macSign.LocalZipFile == "" {
+		catcher.Add(errors.New("local_zip_file cannot be blank"))
+	}
+	if macSign.OutputZipFile == "" {
+		catcher.Add(errors.New("output_zip_file cannot be blank"))
+	}
+	if macSign.ServiceUrl == "" {
+		catcher.Add(errors.New("service_url cannot be blank"))
+	}
+	if macSign.Verify && runtime.GOOS != "darwin" {
+		catcher.Add(errors.New("verify is only supported in macOS"))
+	}
+	if !(macSign.ArtifactType == "" || macSign.ArtifactType == "binary" || macSign.ArtifactType == "app") {
+		catcher.Add((errors.New(`artifact_type needs to be either blank, "binary" or "app"`)))
+	}
+	if macSign.Notarize && macSign.BundleId == "" {
+		catcher.Add(errors.New("if notarization is requested, bundle_id cannot be blank"))
+	}
+
+	return catcher.Resolve()
+}
+
+// Apply the expansions from the relevant task config (including restricted expansion)
+// to all appropriate fields of the macSign.
+func (macSign *macSign) expandParams(conf *internal.TaskConfig) error {
+	var err error
+	if err = util.ExpandValues(macSign, conf.GetExpansionsWithRestricted()); err != nil {
```

Comment:
Please use conf.Expansions; GetExpansionsWithRestricted is something special but also we're planning to remove it.